librz/core/cmd/cmd_print.c

```diff
@@ -6969,7 +6969,13 @@ static void printraw(RzCore *core, int len) {
 	if (!data) {
 		return;
 	}
-	if (rz_io_read_at(core->io, core->offset, data, len)) {
+	if(rz_io_read_at(core->io, core->offset, data, len)){
+		for (int i = 0; i<len; i++){
+			if(data[i]=='\0'){
+				len = i;
+				break;
+			}
+		}
 		rz_print_raw(core->print, core->offset, data, len);
```

Comment:
no. `prz` (print raw zero) is the one that stops at the first `\0` byte, meanwhile `pr` (print raw) should keep printing till the end of the buffer

Reply:
Check how `rz_print_raw` works and modify it to allow to stop the printing when a zero or not. then modify all the calls to this method to either print till the end of the buffer or till the first zero.

Reply:
I wouldn't trust my programming skills, but shouldn't:

https://github.com/rizinorg/rizin/blob/b9851b7c24ab5d3cc2d0192d4b8f8859cad2999c/librz/core/cmd/cmd_print.c#L6979-L6986

really be:
``` c
RZ_IPI RzCmdStatus rz_cmd_print_raw_handler(RzCore *core, int argc, const char **argv) {
        int len = argc > 1 ? rz_num_math(core->num, argv[1]) : core->blocksize;
        if (len < 0) {
                return RZ_CMD_STATUS_ERROR;
        }
        printraw(core, len);
        return RZ_CMD_STATUS_OK;
}
```
I.e., don't try to determine any string length.  Use the block size.

Reply:
definitely you cannot call `strlen`

Reply:
Suree, Will do it

Reply:
Basically, a rough idea of what we can do is in our rz_print_raw function we can add another parameter that basically tells us if we have to stop at the first \0 or not, then modify all the other functions that are calling this method.